### ISSUE TITLE
Cocos2dUpdater: show progress information (ie % download progress)

### ISSUE THREAD

**Issue by LearnCocos2D** [spritebuilder/SpriteBuilder#447]
Updating cocos2d shows a generic progress indicator. However this progress may take quite a long time, quite easily several minutes, possibly hours sometimes. It quickly feels like the app is frozen and perhaps never to finish the update after all.

A label that shows % downloaded or transfer rate or estimated time remaining - anything that tells the user the process did not lock up but instead is still working would be a significant improvement. 

I imagine users are easily tempted to just force quit the app and _not_ update (and possibly deny any future update requests in general) since there's no way of telling when or whether at all the update task will end. What I'm saying is, without improving the progressbar UI users are much less likely to accept the updates which defies the purpose of the feature.

**Comment by NickyWeber** [comment]
I agree: Real progress (% and maybe text of the current step) would be nice to have as well as a cancel button.

In fact it is not downloading anything. I think the unzipping is taking that long. Can you set Cocos2UpdateLogging to 1 in Cocos2Updater.m and see what takes that long in your case?

**Comment by LearnCocos2D** [comment]
Will do. I thought maybe it's just not working from a develop build because version always reads "(null)".

Hmmm I created a new project and it's no longer asking me to update cocos2d. But I got this in the log:

```
2014-05-07 18:31:38.768 SpriteBuilder[20479:303] <Cocos2dUpdater.m:373> [COCO2D-UPDATER] [ERROR] reading SB's cocos2d version file: Error Domain=NSCocoaErrorDomain Code=258 "The file name is invalid."
2014-05-07 18:31:38.768 SpriteBuilder[20479:303] <Cocos2dUpdater.m:376> [COCO2D-UPDATER] [INFO] SpriteBuilder's cocos2d version: (null)
2014-05-07 18:31:38.768 SpriteBuilder[20479:303] <Cocos2dUpdater.m:539> [COCO2D-UPDATER] [INFO] Version file found in Project: 3.0.0
2014-05-07 18:31:38.768 SpriteBuilder[20479:303] <Cocos2dUpdater.m:480> [COCO2D-UPDATER] [INFO] Comparing version - SB: (null) with project: 3.0.0 ...
2014-05-07 18:31:38.769 SpriteBuilder[20479:303] <Cocos2dUpdater.m:501> [COCO2D-UPDATER] [INFO] Testing if there's a .gitmodules file ...
2014-05-07 18:31:38.769 SpriteBuilder[20479:303] <Cocos2dUpdater.m:480> [COCO2D-UPDATER] [INFO] Comparing version - SB: (null) with project: 3.0.0 ...
2014-05-07 18:31:38.769 SpriteBuilder[20479:303] <Cocos2dUpdater.m:501> [COCO2D-UPDATER] [INFO] Testing if there's a .gitmodules file ...
2014-05-07 18:31:38.769 SpriteBuilder[20479:303] <Cocos2dUpdater.m:480> [COCO2D-UPDATER] [INFO] Comparing version - SB: (null) with project: 3.0.0 ...
```

Using latest develop as of right now.

**Comment by NickyWeber** [comment]
Thanks, that's quite interesting, I will have a look at it tomorrow. Maybe something changed regarding the version file. 

EDIT: solved, added some more log info on what went wrong in this case and how to fix it (rerun scripts/BuildDistribution.sh).
